packages/blade/src/components/Button/BaseButton/BaseButton.tsx

```diff
@@ -0,0 +1,259 @@
+      props.lineHeight = buttonTypographyTokens.lineHeights.xsmall;
+      props.minHeight = '28px';
+      props.iconSpacing = makeSpace(theme.spacing[1]);
+      props.spacing = `${makeSpace(theme.spacing[0])} ${makeSpace(theme.spacing[2])}`;
+      props.text = children?.trim().toUpperCase();
+      break;
+    default:
+  }
+
+  if (isDisabled) {
+    const disabledColor = getIn(theme.colors, 'action.background.primary.disabled');
+    const disabledBorderColor = getIn(theme.colors, 'action.border.primary.disabled');
+    props.iconColor = 'action.icon.primary.disabled';
+    props.textColor = 'action.text.primary.disabled';
+    props.color = disabledColor;
+    props.borderColor = disabledBorderColor;
+    props.hoverColor = disabledColor;
+    props.hoverBorderColor = disabledBorderColor;
+    props.activeColor = disabledColor;
+    props.activeBorderColor = disabledBorderColor;
+    props.focusColor = disabledColor;
+    props.focusBorderColor = disabledBorderColor;
+  }
+
+  return props;
+};
+
+const BaseButton = ({
+  variant = 'primary',
+  intent,
+  contrast = 'low',
+  size = 'medium',
+  icon: Icon,
+  iconPosition = 'left',
+  isDisabled = false,
+  isFullWidth = false,
+  onClick,
+  type = 'button',
+  children,
+}: BaseButtonProps): ReactElement => {
+  const { theme, platform } = useTheme();
+  const buttonTypographyTokens = typography[platform];
+  if (!Icon && !children?.trim()) {
+    throw new Error(`[Blade: BaseButton]: Cannot render a BaseButton without an icon or text`);
+  }
+  const {
+    activeBorderColor,
+    activeColor,
+    borderColor,
+    color,
+    minHeight,
+    spacing,
+    focusBorderColor,
+    focusColor,
+    focusRingColor,
+    fontSize,
+    hoverBorderColor,
+    hoverColor,
+    iconColor,
+    iconSize,
+    iconSpacing,
+    lineHeight,
+    text,
+    textColor,
+    borderWidth,
+    borderRadius,
+  } = getProps({
+    buttonTypographyTokens,
+    children,
+    isDisabled,
+    size,
+    theme,
+  });
+  console.log('unused props', {
+    variant,
+    intent,
+    contrast,
+  });
+  return (
+    <StyledBaseButton
+      activeBorderColor={activeBorderColor}
+      activeColor={activeColor}
+      borderColor={borderColor}
+      minHeight={minHeight}
+      spacing={spacing}
+      color={color}
+      disabled={isDisabled}
+      focusBorderColor={focusBorderColor}
+      focusColor={focusColor}
+      focusRingColor={focusRingColor}
+      hoverBorderColor={hoverBorderColor}
+      hoverColor={hoverColor}
+      isFullWidth={isFullWidth}
+      onClick={onClick}
+      type={type}
+      borderWidth={borderWidth}
+      borderRadius={borderRadius}
+    >
+      {Icon && iconPosition == 'left' ? <Icon size={iconSize} color={iconColor} /> : null}
+      {text && (
+        <StyledBaseText
+          lineHeight={lineHeight}
+          fontSize={fontSize}
+          fontWeight="bold"
+          textAlign="center"
+          color={textColor}
```

Comment:
don't we explicitly need to add these properties to the `StyledBaseText` argument? we just have mentioned few of the props there

Reply:
Not sure what you mean here. Its an instance of `BaseText`, we're only extending it with a styled wrapper